Dockerfile

```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM golang:1.15 as builder
+FROM golang:1.16 as builder
```

Comment:
could you update the golang version of the gitlab runner and github action too 🙇

Reply:
This had already been done in a previous PR #387:
https://github.com/DataDog/datadog-operator/blob/5d59e5faa9454a984afdae533b0a5e8303fd1455/.gitlab-ci.yml#L1
https://github.com/DataDog/datadog-operator/blob/5d59e5faa9454a984afdae533b0a5e8303fd1455/.github/workflows/main.yml#L13